src/lib/common/sol-mainloop-impl-riot.c

```diff
@@ -259,78 +108,31 @@ sol_mainloop_impl_run(void)
 void
 sol_mainloop_impl_quit(void)
 {
-    run_loop = false;
+    mainloop_common_run_loop = false;
 }
 
 void *
 sol_mainloop_impl_timeout_add(unsigned int timeout_ms, bool (*cb)(void *data), const void *data)
 {
-    struct timespec now;
-    int ret;
-    struct sol_timeout_riot *timeout = malloc(sizeof(struct sol_timeout_riot));
-
-    SOL_NULL_CHECK(timeout, NULL);
-
-    timeout->timeout = sol_util_timespec_from_msec(timeout_ms);
-    timeout->cb = cb;
-    timeout->data = data;
-    timeout->remove_me = false;
-
-    now = sol_util_timespec_get_current();
-    sol_util_timespec_sum(&now, &timeout->timeout, &timeout->expire);
-    ret = sol_ptr_vector_insert_sorted(&timeout_vector, timeout, timeout_compare);
-    SOL_INT_CHECK_GOTO(ret, != 0, clean);
-    return timeout;
-
-clean:
-    free(timeout);
-    return NULL;
+    return sol_mainloop_impl_common_timeout_add(timeout_ms, cb, data);
```

Comment:
That's a good question. Right now POSIX just adds locking, and even that we could generalize to share more code. We won't support child watch on RIOT, I think, right?